eng/testing/tests.targets

```diff
@@ -136,6 +132,23 @@
       <RunScriptOutputPath>$([MSBuild]::NormalizePath('$(RunScriptOutputDirectory)', '$(RunScriptOutputName)'))</RunScriptOutputPath>
     </PropertyGroup>
 
+    <PropertyGroup Condition="'$(IsXUnitLogCheckerSupported)' == 'true' and
```

Comment:
I see you switched to use a property for consistency with `RunScriptCommand`.  I think with `RunScriptCommand` that was used to represent a single command to run the test whereas with set it's meant to be a list.  IMO it'd be more readable to keep `SetScriptCommands` as an item:

```xml
<ItemGroup Condition="'$(IsXUnitLogCheckerSupported)' == 'true' and
                      '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
                      $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '$(NETCoreAppCurrentVersion)'))">
  <SetScriptCommands Condition="'$(TargetOS)' == 'windows'" Include="set __TestArchitecture=$(TargetArchitecture);set __IsXUnitLogCheckerSupported=1" />
  <SetScriptCommands Condition="'$(TargetOS)' != 'windows'" Include="export __TestArchitecture=$(TargetArchitecture);export __IsXUnitLogCheckerSupported=1" />
</ItemGroup>
```

PS: I noticed this when asking the questions: who else sets `SetScriptCommand`?  Will it be semi-colon, or new-line delimited?

Reply:
I think we should keep `SetScriptCommands` as an `ItemGroup`. It's also used by `tests.browser.targets`, `tests.wasi.targets`, and the `Directory.Build.targets` in `src/mono/sample`. It might not cause any issues whatsoever, but I'm scared it could potentially collide with them in the future, as they all treat it as a list in an `ItemGroup`.

Reply:
>  who else sets SetScriptCommand?

In tests.targets? No one else was using SetScriptCommand. I added its use first use here: https://github.com/dotnet/runtime/pull/94868/files#diff-ea069e899aadcfcf3dcabecd88dbd80e12d6cb331de055fba7f49c9ef9a329adR72

I also added its first use in the runner templates used by libraries: [cmd](https://github.com/dotnet/runtime/pull/94868/files#diff-0bcef275f5c5df0bbd24b8db87c2d4c65cecb2770abe0463dac97074d19f6d5aR71-R72) and [sh](https://github.com/dotnet/runtime/pull/94868/files#diff-f7fb37056cce8903c2a743e9d9f9d6b9eee30041677827b9e0fa51d5f9378012R167-R168).

As Ivan mentioned, it is already used in many other places.

Just in case, that's why I made sure to precede the value of the properties with `$(SetScriptCommand)` in case there is a pre-existing value. I used a similar example from lines 99 and 100.

I also decided to follow the same pre-existing pattern of setting the additional values inside a property (with the name in singular) where I prefix the value with the same property, then include it to the item in a new itemgroup (with the name in plural), and then use the itemgroup in the task attribute (see RunScriptCommand and RunScriptCommands).

Reply:
> Will it be semi-colon, or new-line delimited?

I have noticed both uses in several files. Seems to be the same, although I got a suggestion in my previous PR (in a chat conversation, not as PR feedback) to use endlines instead of semicolons. Both seem to work fine.